concourse/tasks/run_behave_on_ccp_cluster.yml

```
@@ -16,28 +16,29 @@ run:
 
     # TODO: ask CCP maintainers for a feature to do this for us
     scp cluster_env_files/hostfile_all mdw:/tmp
-    cp cluster_env_files/hostfile_all /tmp
 
     # Install patchelf. We need to SSH as root, hence the use of
     # cluster_env_files.
     ssh -t ccp-$(cat cluster_env_files/terraform/name)-0 "sudo bash -c '
         source /home/gpadmin/gpdb_src/concourse/scripts/common.bash
         install_python_hacks
-        install_python_requirements /home/gpadmin/gpdb_src/gpMgmt/requirements-dev.txt
+    '"
+
+    ssh -t mdw "
+        source /home/gpadmin/gpdb_src/concourse/scripts/common.bash
+        install_python_requirements_on_multi_host /home/gpadmin/gpdb_src/gpMgmt/requirements-dev.txt
 
         # Enable coverage.py on all hosts. (This modifies greenplum_path.sh and must
         # come before the source below.)
         setup_coverage /home/gpadmin/gpdb_src
-    '"
 
-    ssh mdw "
-        bash /home/gpadmin/gpdb_src/concourse/scripts/run_behave_test.sh \"$BEHAVE_FLAGS\"
-
-        source /home/gpadmin/gpdb_src/concourse/scripts/common.bash
-        # Since mdw is not a concourse container it will not have the output
-        # directory coverage created, as is done for demo_cluster tests
-        mkdir -p  /home/gpadmin/coverage
-        collect_coverage \"$TEST_NAME\"
+        $CUSTOM_ENV bash /home/gpadmin/gpdb_src/concourse/scripts/run_behave_test.sh \"$BEHAVE_FLAGS\"
     "
 
-    scp -r mdw:/home/gpadmin/coverage/* ./coverage/
+    # collect coverage
+    while read -r host; do
+        scp -r "$host:/tmp/coverage/*" ./coverage/
+    done < cluster_env_files/hostfile_all
+
+    source ./gpdb_src/concourse/scripts/common.bash
+    tar_coverage \"$TEST_NAME\"
```

Why are the quotes escaped here?
```sh
$ gsutil ls 'gs://pivotal-gpdb-cli-coverage-dev/gpdb-cm-mark-s-cli-code-coverage-full/39d86ecb14e778bcdd096fda08e7628115753da2/*.tar'
gs://pivotal-gpdb-cli-coverage-dev/gpdb-cm-mark-s-cli-code-coverage-full/39d86ecb14e778bcdd096fda08e7628115753da2/"gpactivatestandby".tar
gs://pivotal-gpdb-cli-coverage-dev/gpdb-cm-mark-s-cli-code-coverage-full/39d86ecb14e778bcdd096fda08e7628115753da2/"gpaddmirrors".tar
gs://pivotal-gpdb-cli-coverage-dev/gpdb-cm-mark-s-cli-code-coverage-full/39d86ecb14e778bcdd096fda08e7628115753da2/"gpcheckcat".tar
...
```

---

Most likely a hold-over from old code. Will remove them.

---

It seems like, if the quotes are removed, we'll end up clobbering the tar file that was pushed by another test? Now that we've moved to the "standard" CCP cluster worker permissions, will that error out and let us know there's a problem, or will it silently succeed?

---

Right, the ccp and demo cluster coverage tar files will have the same name and collide. I added a fixup to name them ${TEST_NAME}_demo and ${TEST_NAME}_ccp. Thanks for calling that out - its easy to get a bit lost in the weeds with all this.